### DISCUSSION TITLE
Guidance for how to access database directly from desktop clients

### DISCUSSION THREAD

**Discussion by sam-wheat** [dotnet/SqlClient#1553] [General]
Client wants me to write a WPF desktop app that will run on their internal LAN.   I want users of my app to use sqlclient to access the database directly versus a web API.   

The concern is that users will be able to use a CLI or SQL Management Studio to access the database outside of the app.  Is there a way for me to control access to the database such that only users of my app can use it?

Thanks

**Comment by Kaur-Parminder** [comment]
> Is there a way for me to control access to the database such that only users of my app can use it?

@sam-wheat depending on your needs you can choose type of login SQL Server supports (Windows account or as a member of a Windows group, or the login can be a SQL Server login that exists only in SQL Server) see https://docs.microsoft.com/en-us/sql/t-sql/lesson-2-configuring-permissions-on-database-objects?view=sql-server-ver15 
and you can choose type of roles or permission you want to assign to users see SQL Server Concepts https://docs.microsoft.com/en-us/sql/relational-databases/security/authentication-access/application-roles?view=sql-server-ver15 (Database level roles, server level roles or application level roles).
